components/dashboard/src/app/WebsocketClients.tsx

```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useListenToWorkspacesWSMessages } from "../data/workspaces/queries";
+
+// This component is intended to setup any app-wide websocket client subscriptions
+// It doesn't render anything directly
+export const WebsocketClients = () => {
```

Comment:
Am open to any better names here, `WebsocketClients` feels pretty generic, but it's intended to house any listeners we need to setup for incoming websocket messages 🤔

Reply:
I'd directly call `useListenToWorkspaceWSMessages` where we need that instead of adding a generic indirection here.

Reply:
We need it at a fairly high level of the app component hierarchy, after we've established you're a logged in user, which is what led me to using a component to wrap it, and then adding that component inside of our routes renderer. 

I initially had it so I called `useListenToWorkspaceWSMessages()` directly in the `<AppRoutes/>` component, but there's a fair amount of logic (like the various location checks that render alternate flows/views) that happens in that component before it renders that I was hoping to setup the listener afterwards. Aside from reorganizing the `<AppRoutes/>` component so there's a 2nd layer of component there, this was my best idea. It's probably ok to just do it there instead, so I can give that a shot, was a little reluctant to mess with that logic too much since it renders the StartWorkspace flow (i'm not too familiar with it yet), and a few other things outside of the Router.